### PR TITLE
[xregexp] Add types for `union` function to accept either `string[]` and `RegExp[]`

### PR DESCRIPTION
**Pull request by sigo**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/xregexp/index.d.ts [hunk 1]
```diff
@@ -83,7 +84,7 @@ declare namespace OuterXRegExp {
     function uninstall(options: Object): void;
     function uninstall(options: string): void;
 
-    function union(patterns: string[], flags?: string): RegExp;
+    function union(patterns: string[] | RegExp[], flags?: string): RegExp;
```
Comment (RyanCavanaugh) [inline types/xregexp/index.d.ts]: Seems like this is actually `(string | RegExp)[]` - it need not be uniform. The docs show the example `XRegExp.union(['a+b*c', /(dogs)\1/, /(cats)\1/], 'i');`

Comment (sigo) [inline types/xregexp/index.d.ts]: @RyanCavanaugh Sorry, I don't understand _it need not be uniform_ phrase. Could you tell it in other words?

Should I only change `string[] | RegExp[]` entry to the `(string | RegExp)[]`?

Comment (RyanCavanaugh) [inline types/xregexp/index.d.ts]: `string[] | RegExp[]` means you can pass `['hello', 'world']` or `[/hello/, /world/]` but not `['hello', /world/]`. `(string | RegExp)[]` allows `['hello', /world/]` which appears to be an allowed value based on the examples

**Review by jfahrenkrug — APPROVED** [review]

**Comment by jfahrenkrug** [comment]
@sigo Nice work, thanks!

**Pull request merged**